packages/network-controller/src/create-auto-managed-network-client.ts

```diff
@@ -0,0 +1,172 @@
+import { createNetworkClient, NetworkClient } from './create-network-client';
+import { BlockTracker, NetworkClientConfiguration, Provider } from './types';
+
+/**
+ * Represents a proxy object which wraps a target object. As a proxy, it allows
+ * for accessing and setting all of the properties that the target object
+ * supports, but also supports an extra propertyName (`__target__`) to access
+ * the target itself.
+ *
+ * @template Type - The type of the target object. It is assumed that this type
+ * will be constant even when the target is swapped.
+ */
+export type ProxyWithAccessibleTarget<TargetType> = TargetType & {
+  __target__: TargetType;
```

Comment:
Nit: Could we use `REFLECTIVE_PROPERTY_NAME` here rather as well?

Reply:
Yup! Updated in [5f646b4](https://github.com/MetaMask/core/pull/1439/commits/5f646b42ebfd41122b94bd7cfed0a634c0e31b84).